syntax = "proto2";

// protoc -I=./ --cpp_out=./  raft.proto

package echo;

option cc_generic_services = true;

message Request {
  required int64 sequence_num = 1;
  required bytes request = 2;
};

message Response {
  required int64 sequence_num = 1;
  required bool success = 2;
  optional string server_id = 3;
  optional string response = 4;
};

service ProtoEchoService {
  rpc Execute(Request) returns (Response);
}
